core/src/components/cat-tooltip/cat-tooltip.tsx

```diff
@@ -0,0 +1,104 @@
+import {Component, h, Host, Prop} from '@stencil/core';
+import {autoUpdate, computePosition, flip, offset, Placement} from "@floating-ui/dom";
+
+@Component({
+  tag: 'cat-tooltip',
+  styleUrl: 'cat-tooltip.scss',
+  shadow: true,
+})
+export class CatTooltip {
+  private tooltip?: HTMLElement;
+  private trigger?: Element;
+  private showTimeout: NodeJS.Timeout;
```

Comment:
The type is wrong. it should be number. You will get the correct type, if you use the proper window.setTimeout and window.clearTimeout. These are coming from lib dom